src/compiler/declarationEmitter.ts

```diff
@@ -1367,6 +1367,11 @@ namespace ts {
         }
 
         function writeVariableStatement(node: VariableStatement) {
+            // If binding pattern doesn't have name, then there is nothing to be emitted for declaration file i.e. const [,] = [1,2].
+            if (node.declarationList.declarations.length && isBindingPattern(node.declarationList.declarations[0].name) &&

```

Comment:
This needs to be a recursive check, as binding patterns can nest infinitely, and of not just omitted expressions, but omitted expressions or binding patterns containing only the aforementioned.

Also, a test for that:
```ts
// @declaration: true
export let [,,[,[],,[],]] = undefined as any;
```